docs/topic-guides/workflows/workflow-file.rst

```diff
@@ -0,0 +1,362 @@
+.. _workflow-definition-file-topic-guide:
+
+The Renku Workflow File
+=======================
+
+If your data processing workflow has many steps, you might find it convenient to
+create a workflow definition file. A workflow file help organize your data
+processing flow, making it easier to execute portions of the pipeline at a time.
+A workflow file also makes your code easier for your collaborators to
+understand!
+
+
+Adding a Workflow File to your Project
+--------------------------------------
+
+To create a workflow definition file in your project, create a file called
+``workflow.yml``. 
+
+You may have more than one workflow file in your project! Workflow files may be
+named however you like, as long as they end with the `.yml` or `yaml` file
+extension.
+
+.. note:: **Do you already have a workflow you created on the Renku CLI that you'd like to convert to a Workflow File?**
+    You can export it! ``renku workflow export <workflow-name> --format renku --output workflow.yml``
+    (Tip: Find your workflow's name using ``renku workflow ls``)
+
+
+Defining a Basic Workflow File
+------------------------------
+
+There are a few options for how you may define your workflow.
+
+In the simplest version of the Renku workflow file, the command, inputs and
+outputs are simply listed, as in the example below:
+
+.. code-block:: yaml
+
+    name: flights-processing-pipeline
+    steps:
+      filter:
+        command: python src/filter_flights.py data/flight-data/2019-01-flights.csv.zip data/output/flights-filtered.csv
+        inputs:
+          - src/filter_flights.py
+          - data/flight-data/2019-01-flights.csv.zip
+        outputs:
+          - data/output/flights-filtered.csv
+
+.. note:: Note that the script being run is always listed in the ``inputs`` section (here, ``src/filter_flights.py``).
+
+This workflow file defines the workflow's name and a sequence of steps. This
+file only includes one step, which is named ``filter``. Within the ``filter``
+step, list the ``command`` to run, and then we tell Renku which parts of this
+command are ``inputs`` and ``outputs`` by copying those paths into the relevant
+sections.
+
```

Comment:
It's good to have a note or warning to say that the workflow file itself is a YAML file but with basic format. Something like the following:

```
Renku uses basic YAML syntax for workflow definition files. 
Users shouldn't use advanced YAML syntax like anchors, aliases, schema, etc. since the behavior is undefined. 
Moreover, in future we will implement a customized YAML parser that won't allow these features.
```

Reply:
Good point. I added this as a note at the bottom of the Templating section, since that's where I'd guess users might look for info about syntax shortcuts.